### PR TITLE
feat!: support async resolver & compatible with unplugin-vue-components, close #43

### LINKED ISSUE #43: How can I use AntDesignVueResolver in unplugin-auto-import?
How can I use AntDesignVueResolver in unplugin-auto-import

### PR DESCRIPTION
**Pull request by sxzz**
- chore: format package.json
- fix: generate dts when import *
- feat!: support async resolver & compatible with unplugin-vue-components

⚠️ This is a breaking change.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,17 +101,10 @@
     "ts-jest": "^27.0.7",
     "tsup": "^5.10.0",
     "typescript": "^4.5.2",
+    "unplugin-vue-components": "^0.17.2",
```
Comment (antfu) [inline package.json:104]: Can we avoid directly relying on it?

Comment (sxzz) [reply package.json:104]: But we have to import its type `ComponentResolver`

Comment (sxzz) [reply package.json:104]: Or we just make it optional?

File: playground/vite.config.ts [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 import AutoImport from '../src/vite'
+import { componentResolver } from '../src'
 
 export default defineConfig({
   plugins: [
     Vue(),
     Inspect(),
     AutoImport({
       imports: ['vue', '@vueuse/core'],
+      resolvers: [componentResolver(ElementPlusResolver())],
```
Comment (antfu) [inline playground/vite.config.ts:14]: Could we possibly make it just work without the converting function?

Comment (sxzz) [reply playground/vite.config.ts:14]: This is a bit difficult, because components and directives need to be handled separately.

Comment (sxzz) [reply playground/vite.config.ts]: @antfu Done.

**Comment by sxzz** [comment]
Wait a second

**Pull request merged**